hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -369,8 +373,15 @@ private void runTasklet(TaskletTracker t) {
                 }
                 progressTracker.mergeWith(result);
             } catch (Throwable e) {
-                logger.warning("Exception in " + t.tasklet, e);
-                t.executionTracker.exception(new JetException("Exception in " + t.tasklet + ": " + e, e));
+                // No need to print collected exceptions when user cancels the job.
+                // We can use instanceof here, because it is 'cold' code.
+                if (e instanceof QueryException && ((QueryException) e).getCode() == CANCELLED_BY_USER) {
+                    logger.warning("Job was cancelled by the user.");
```

Comment:
I didn't test it, but will it print this warning once for each tasklet in the job?

Reply:
Correct